### PR TITLE
Implementation of From<T> for Collection<T> and CollectionWithId<T>

### PR DESCRIPTION
**Pull request by woshilapin**
Often, we want to create a `Collection` or a `CollectionWithId` out of only one element. Implementation of `From<T> for Collection<T>` and `From<T> for CollectionWithId<T>` should help with that.

### CODE REVIEW

File: src/collection.rs [hunk 1]
```diff
@@ -361,6 +378,38 @@ pub struct CollectionWithId<T> {
     id_to_idx: HashMap<String, Idx<T>>,
 }
 
+/// Creates a `CollectionWithId` from one element.
+///
+/// This cannot fail since there will be a unique identifier in the collection
```
Comment (datanel) [inline src/collection.rs]: you are right, that's why you can use `CollectionWithId::new(vec![object]).unwrap()` without hesitation

Comment (woshilapin) [reply src/collection.rs]: Done in 1e0da0a40a1a1207612bbd045ae491625908c34f.

**Comment by datanel** [comment]
yo have to rebase

**Comment by woshilapin** [comment]
I just rebased, it should be good now.

**Pull request merged**